pymc/data.py

```diff
@@ -148,6 +149,24 @@ def __str__(self):
         return "Minibatch"
 
 
+def first_inputs(r):
+    if not r.owner:
+        return
+
+    inputs = r.owner.inputs
+
+    if not inputs:
+        return
+
+    first_input = inputs[0]
+    yield first_input
+    yield from first_inputs(first_input)
+
+
+def has_random_ancestor(r):
+    return any(isinstance(i, RandomGeneratorSharedVariable) for i in first_inputs(r))
```

Comment:
Check: https://github.com/pymc-devs/pymc/blob/fa43eba8d682fbf4039bbd1c228c940c872e2d5d/pymc/logprob/utils.py#L133-L150

Reply:
Thanks for pointing out. I've changed to use this